### PR TITLE
feat: remove dups and improve performance (namu.wiki)

### PR DESCRIPTION
**Pull request by seia-soto**
This PR is from:

- https://github.com/List-KR/List-KR/issues/236#issuecomment-1081011725

Appling:

```css
namu.wiki#?#table:not([class][style]):has(+ div[class="wiki-paragraph"]:empty)
```

image

### CODE REVIEW

**Comment by seia-soto** [comment]
Outdated. Will update the rule soon.

**Review by piquark6046 — CHANGES_REQUESTED** [review]
Please make sure that your new rules must work on a mobile device.

**Comment by seia-soto** [comment]
@piquark6046 Unfortunately, I was unable to resolve the problem both on iOS and Android. Is there a specific documentation about this issue?

**Comment by TetraTheta** [comment]
I think it is still broken. It doesn't block 'Powerlink' advertisement section at the bottom of the page. Page performance is highly improved, though. No visible lag, No slowdown etc.

`<div>` that has random string as class, suspective for space for advertisement still shows in some page. I don't know if this is from List-KR or not.

Tested filter URL is `https://raw.githubusercontent.com/seia-soto/List-KR/namu.wiki/20220330-improvement/filter.txt`, replacing List-KR (from List-KR repo)

Environment: AdGuard Windows 7.9 + Windows 11 + Chrome 99.0.4844.84

<details><summary>Screenshots</summary>
<p>

</p>
<p>

</p>
<p>&nbsp;</p>
<p></p>
<p></p>
</details>

I hope this information helps. If not, sorry for the useless information.

**Comment by seia-soto** [comment]
> I think it is still broken. It doesn't block 'Powerlink' advertisement section at the bottom of the page. Page performance is highly improved, though. No visible lag, No slowdown etc.
> 
> `<div>` that has random string as class, suspective for space for advertisement still shows in some page. I don't know if this is from List-KR or not.
> 
> Tested filter URL is `https://raw.githubusercontent.com/seia-soto/List-KR/namu.wiki/20220330-improvement/filter.txt`, replacing List-KR (from List-KR repo)
> 
> Environment: AdGuard Windows 7.9 + Windows 11 + Chrome 99.0.4844.84
> Screenshots
> 
> I hope this information helps. If not, sorry for the useless information.

The rule can be outdated. Here are some rules that can be used as alternatives:

```css
namu.wiki#?#div:not([class]):has(> div:not([class=""]) + .wiki-paragraph:empty)
namu.wiki#?#div:not([class]):has(> table:not([class=""]) + .wiki-paragraph:empty)
namu.wiki#?#div:has(> table + .wiki-paragraph:empty)
```

**Comment by TetraTheta** [comment]
> The rule can be outdated. Here are some rules that can be used as alternatives:
> 
> ```css
> namu.wiki#?#div:not([class]):has(> div:not([class=""]) + .wiki-paragraph:empty)
> namu.wiki#?#div:not([class]):has(> table:not([class=""]) + .wiki-paragraph:empty)
> namu.wiki#?#div:has(> table + .wiki-paragraph:empty)
> ```

I can still see Powerlink section and blank div. I removed your List-KR filter and added those lines to user rules.  
So, There is no List-KR present at my current AdGuard environment except for those lines.

TL;DR - Those lines didn't work too.

**Comment by seia-soto** [comment]
> > The rule can be outdated. Here are some rules that can be used as alternatives:
> > ```css
> > namu.wiki#?#div:not([class]):has(> div:not([class=""]) + .wiki-paragraph:empty)
> > namu.wiki#?#div:not([class]):has(> table:not([class=""]) + .wiki-paragraph:empty)
> > namu.wiki#?#div:has(> table + .wiki-paragraph:empty)
> > ```
> 
> I can still see Powerlink section and blank div. I removed your List-KR filter and added those lines to user rules. So, There is no List-KR present at my current AdGuard environment except for those lines.
> 
> TL;DR - Those lines didn't work too.

I have no idea for now then. Unable to reproduce your situation.

**Comment by piquark6046** [comment]
@TetraTheta 
Do you use a VPN?

---

@seia-soto 
I think that remote access (e.g. TeamViewer) is a good idea to reproduce.

**Comment by seia-soto** [comment]
> @TetraTheta Do you use a VPN?
> 
> @seia-soto I think that remote access (e.g. TeamViewer) is a good idea to reproduce.

I think so. @TetraTheta , if you prefer, please contact me using encrypted mail from following support page with your time availibility.

- https://github.com/List-KR/List-KR/issues/223

Otherwise, save the page you're seeing and send it in zipped state to me.

**Comment by seia-soto** [comment]
@piquark6046 The update process of this PR is completed. Mobile availability is still not confirmed.

**Comment by seia-soto** [comment]
Noticed that I can reproduce the error of the rule on macOS Safari arm64, 15.4(17613.1.17.1.6) using AdGuard Safari Extension just like mobile was.

## Logs

image

image

**Review by piquark6046 — APPROVED** [review]

**Comment by seia-soto** [comment]
I created new issue related to this problem:

- https://github.com/AdguardTeam/AdGuardForSafari/issues/698
- https://github.com/AdguardTeam/AdguardForiOS/issues/1968
